### DISCUSSION TITLE
Simple schema validation

### DISCUSSION THREAD

**Discussion by zachfi** [cuelang/cue#876] [Q&A]
With the following cue, I'm receiving an error when I run `cue vet modules.cue modules.yaml` that I don't understand.

```
[...#GitRef]
#GitRef: {
  name: string
  git: string
  ref: string
}
```

I'm trying to ensure that a yaml file conforms to the above definition, but I receive the following error: `field #GitRef not allowed`.  It does appear that the yaml gets validated to conform to this spec, but the error is confusing since I was under the impression that `#` means the field won't need to exist in the document.  What am I overlooking?

**Comment by myitcv** [comment]
@xaque208 welcome to the CUE project!

Please can you provide a complete example of what you are trying? That will help to explain what is going on.

**Reply by zachfi** [reply]
Thank you for the reply.  Here is the complete example of what I'm working with.

```bash
❯ cat modules.cue
[...#GitRef]
#GitRef: {
  name: string
  git: string
  ref: string
}

❯ cat test.yaml
---
- name: account
  git: git://github.com/xaque208/puppet-account.git
  ref: 1.2.0

❯ cue vet modules.cue test.yaml -v
field #GitRef not allowed
zsh: exit 1     cue vet modules.cue test.yaml -v
```

**Reply by myitcv** [reply]
@xaque208 this looks like a bug to me because if instead of vetting a list we vet individual values we essentially have the scenario described in https://github.com/cuelang/cue/issues/743. I'll create an issue off the back of this.

**Reply by myitcv** [reply]
https://github.com/cuelang/cue/issues/879

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/876.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
